File: src/main/java/ast/Sibling.java
package ast;

import tree.TargetTreeNode;

import java.util.ArrayList;
import java.util.List;

public class Sibling implements ASTNode {

	List<ASTNode> siblings;
	
	
	public Sibling(List<ASTNode> siblings) {
		super();
		this.siblings = siblings;
	}


	/// TODO -throw exception not more nodes , move right
	///  TODO - change list to other 
	@Override
	public Object execute(TargetTreeNode targetTreeNode) {
		List<TargetTreeNode> answer = new ArrayList<>();
		TargetTreeNode current = targetTreeNode;
		int i = 0;
		for( ASTNode node: siblings ) {
			if (i++ != 0) {
				current = current.moveRight();
			}
			List<TargetTreeNode> temporal = (List<TargetTreeNode>) node.execute(current);
			if (temporal.isEmpty()) {
				return new ArrayList<>();
			}
			answer.addAll(temporal);
		}
		return answer;
	}

}

File: src/main/java/ast/Sibling.kt
package ast

import tree.TargetTreeNode

class Sibling(var siblings: List<ASTNode>) : ASTNode {
    // / TODO -throw exception not more nodes , move right
// /  TODO - change list to other
    override fun execute(targetTreeNode: TargetTreeNode): Any? {
        val answer = mutableListOf<ASTNode>()
        var current = targetTreeNode
        var i = 0
        for (node in siblings) {
            if (i++ != 0) {
                current = current.moveRight()
            }
            val temporal = node.execute(current) as List<ASTNode>
            if (temporal.isEmpty()) {
                return ArrayList<Any>()
            }
            answer.addAll(temporal)
        }
        return answer
    }
}
